cli/cluster/cluster.go

```diff
@@ -92,39 +95,48 @@ func collectTarantoolConfig(collectors libcluster.CollectorFactory,
 		} else {
 			network, address = connect.ParseBaseURI(endpoint.Uri)
 		}
-
+		url := fmt.Sprintf("%s://%s", network, address)
 		if endpoint.Params.Transport == "" || endpoint.Params.Transport != "ssl" {
 			opts = append(opts, tarantoolOpts{
+				addr: url,
+				dialer: tarantool.NetDialer{
+					Address:  url,
+					User:     endpoint.Login,
+					Password: endpoint.Password,
+				},
 				opts: tarantool.Opts{
-					User:       endpoint.Login,
-					Pass:       endpoint.Password,
-					Transport:  endpoint.Params.Transport,
 					SkipSchema: true,
 				},
-				addr: fmt.Sprintf("%s://%s", network, address)})
+			})
 		} else {
 			opts = append(opts, tarantoolOpts{
+				addr: url,
+				dialer: tlsdialer.OpenSSLDialer{
+					Address:     url,
+					User:        endpoint.Login,
+					Password:    endpoint.Password,
+					SslKeyFile:  endpoint.Params.SslKeyFile,
+					SslCertFile: endpoint.Params.SslCertFile,
+					SslCaFile:   endpoint.Params.SslCaFile,
+					SslCiphers:  endpoint.Params.SslCiphers,
+				},
 				opts: tarantool.Opts{
-					User:      endpoint.Login,
-					Pass:      endpoint.Password,
-					Transport: endpoint.Params.Transport,
-					Ssl: tarantool.SslOpts{
-						KeyFile:  endpoint.Params.SslKeyFile,
-						CertFile: endpoint.Params.SslCertFile,
-						CaFile:   endpoint.Params.SslCaFile,
-						Ciphers:  endpoint.Params.SslCiphers,
-					},
 					SkipSchema: true,
 				},
-				addr: fmt.Sprintf("%s://%s", network, address),
 			})
 		}
 	}
 
 	var connectionErrors []error
 	cconfig := libcluster.NewConfig()
 	for _, opt := range opts {
-		conn, err := tarantool.Connect(opt.addr, opt.opts)
+		ctx := context.Background()
+		if opt.opts.Timeout > 0 {
+			var cancel context.CancelFunc
+			ctx, cancel = context.WithTimeout(ctx, opt.opts.Timeout)
+			defer cancel()
+		}
+		conn, err := tarantool.Connect(ctx, opt.dialer, opt.opts)
```

Comment:
Newlines in the go-way.
```suggestion

		conn, err := tarantool.Connect(ctx, opt.dialer, opt.opts)
		if err != nil {
```